projection/core/src/main/scala/com/lightbend/lagom/internal/projection/ProjectionRegistryActor.scala

```diff
@@ -4,128 +4,186 @@
+  val readConsistency: ReadConsistency = ReadLocal
+
+  // (a) Replicator contains data of all workers
+  private val RequestedStatusDataKey: LWWMapKey[WorkerKey, Status] =
+    LWWMapKey[WorkerKey, Status]("projection-registry-desired-status")
+  private val ObservedStatusDataKey: LWWMapKey[WorkerKey, Status] =
+    LWWMapKey[WorkerKey, Status]("projection-registry-observed-status")
+  private val NameIndexDataKey: LWWMapKey[WorkerKey, WorkerCoordinates] =
+    LWWMapKey[WorkerKey, WorkerCoordinates]("projection-registry-name-index")
+  replicator ! Subscribe(RequestedStatusDataKey, self)
+  replicator ! Subscribe(ObservedStatusDataKey, self)
+  replicator ! Subscribe(NameIndexDataKey, self)
+
+  // (b) Keep a local copy to simplify the implementation of some ops
+  var requestedStatusLocalCopy: Map[WorkerKey, Status] = Map.empty[WorkerKey, Status]
+  var observedStatusLocalCopy: Map[WorkerKey, Status]  = Map.empty[WorkerKey, Status]
+
+  var nameIndexLocalCopy: Map[WorkerKey, WorkerCoordinates] = Map.empty[WorkerKey, WorkerCoordinates]
+
+  // (c) Actor indices contain only data of workers running locally
+  var actorIndex: Map[WorkerKey, ActorRef] = Map.empty[WorkerKey, ActorRef]
   // required to handle Terminate(deadActor)
-  var actorReverseIndex: Map[ActorRef, WorkerMetadata] = Map.empty[ActorRef, WorkerMetadata]
+  var reversedActorIndex: Map[ActorRef, WorkerKey] = Map.empty[ActorRef, WorkerKey]
 
   override def receive: Receive = {
-    case RegisterProjection(streamName, projectionName, workerName) =>
-      val metadata = WorkerMetadata(streamName, projectionName, workerName)
-      // when registering a projection worker, we default to state==enabled
-      val writeMajority = WriteMajority(timeout = 5.seconds)
-      replicator ! Update(DataKey, PNCounterMap.empty[WorkerMetadata], writeMajority)(
-        //TODO: read the default state from a desired _initial state_
-        // https://github.com/lagom/lagom/issues/1744
-        _.increment(node, metadata, 1)
-      )
-      // keep track and watch
-      actorIndex = actorIndex.updated(metadata, sender)
-      actorReverseIndex = actorReverseIndex.updated(sender, metadata)
+
+    case RegisterProjectionWorker(coordinates) =>
+      log.debug(s"Registering worker $coordinates to [${sender().path.toString}]")
+      // keep track
+      val workerKey: WorkerKey = coordinates.asKey
+      updateLWWMapForNameIndex(workerKey, coordinates)
+      actorIndex = actorIndex.updated(workerKey, sender())
+      reversedActorIndex = reversedActorIndex.updated(sender, workerKey)
+      // when worker registers, we must reply with the requested status (if it's been set already, or Started if not).
+      // TODO: parameterize the default initial status
+      val requestedStatus = requestedStatusLocalCopy.getOrElse(workerKey, Started)
+      log.debug(s"Setting requested status [$requestedStatus] on worker $workerKey [${sender().path.toString}]")
+      sender ! requestedStatus
```

Comment:
Assuming the WorkerHolder is not present on the `requestedStatusLocalCopy` map. 

After telling the WorkerHolder to start we are not updating the replicated data with this new information.

Reply:
> After telling the WorkerHolder to start we are not updating the replicated data with this new information.

Because we must not.

The `requested` is what we want. The worker holder must never decide what the user wants. Only the user can update what the user wants. The worker, OTOH, will update what the actual status is, which the user may _observe_.

Reply:
I don't think this assertion is correct. 

We are requesting it to start. 
```
val requestedStatus = requestedStatusLocalCopy.getOrElse(workerKey, Started)
log.debug(s"Setting requested status [$requestedStatus] on worker $workerKey [${sender().path.toString}]")
```

If we transpose it to the current Lagom behaviour, all projections are requested to start on bootstrap. I don't think that the restriction that `requested` is only when a user requested it is useful. 

Later, when converting the ddate to `State` we are have the following:

```
Worker(
  coordinates.tagName,
  coordinates.asKey,
  desiredStatusLocalCopy.getOrElse(workerKey, Stopped),
  observedStatusLocalCopy.getOrElse(workerKey, Stopped)
)
```

The result of this will be that we will see a worker with the requested state set to `Stopped` and observed state set to `Started`. And we are going to ask ourselves, what is wrong here? Why this thing is not stopping if the requested state is Stopped? 

The reason is that we requested it internally to start, but we didn't register that fact.

Reply:
This was discussed and we will introduce an initial state that should be used when starting a new worker and when reporting its state. Unless ddata propagates and it gets changed.